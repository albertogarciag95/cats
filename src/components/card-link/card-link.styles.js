import styled from 'styled-components';

export const LinkText = styled.a`
  margin-top: 10px;
  color: blue;
  cursor: pointer;
  heigth: 30px;
  ${props => {
    if(!props.isShown) {
      return `visibility: hidden`;
    }
  }}
`;

export const CardPopup = styled.div`
  padding: 20px;
  background-color: #0ea5c8;
  opacity: .9;
  position: absolute;
  min-width: 200px;
  max-width: 400px;
  top: 100%;
`;

export const CardPopupButton = styled.button`
  display: block;
  background-color: blue;
  color: white;
  border-radius: 30px;
  width: 80px;
  margin-top: 20px;
`