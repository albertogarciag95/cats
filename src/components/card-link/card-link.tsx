import React, { useEffect, useState } from 'react';
import { 
  LinkText, 
  CardPopup,
  CardPopupButton
} from './card-link.styles';

type CardLinkProps = {
  description: string,
  maxDescriptionLength: number
};

export const CardLink = ({ description, maxDescriptionLength }: CardLinkProps) => {
  const [isPopupShown, showPopup] = useState(false);

  useEffect(() => {
    showPopup(false)
  }, [description])

  return (
    <>
      <LinkText 
        isShown={description.length > maxDescriptionLength}
        onMouseEnter={() => showPopup(true)}>
          Read more
      </LinkText>
      {isPopupShown && (
        <CardPopup>
          <span>{description}</span>
          <CardPopupButton onClick={() => showPopup(false)}>OK</CardPopupButton>
        </CardPopup>
      )}
    </>
  );
}