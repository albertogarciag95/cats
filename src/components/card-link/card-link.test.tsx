import renderer from 'react-test-renderer';
import Adapter from "enzyme-adapter-react-16";
import { shallow, configure } from 'enzyme';

import { CardLink } from './card-link';

configure({ adapter: new Adapter() });

const props = {
  description: 'test',
  maxDescriptionLength: 100
}

test('CardLink match snapshot', () => {
  const component = renderer.create(
    <CardLink {...props}/>
  );
  let cardlink = component.toJSON();
  expect(cardlink).toMatchSnapshot();
});

test('Show popup with description onMouseEnter', () => {
  const component = shallow(<CardLink {...props}/>)
  component.find('LinkText').props().onMouseEnter({} as any);
  expect(component.find('CardPopup').exists()).toBe(true);
  expect(component.find('span').text()).toBe('test');
});

test('Hide popup with description onMouseLeave', () => {
  const component = shallow(<CardLink {...props}/>)
  component.find('LinkText').props().onMouseEnter({} as any);
  component.find('CardPopupButton').props().onClick()
  expect(component.find('CardPopup').exists()).toBe(false);
});

test('Link text is not shown', () => {
  const component = shallow(<CardLink {...props}/>)
  expect(component.find('LinkText').props().isShown).toBe(false);
});

test('Link text shows', () => {
  const component = shallow(<CardLink {...props} description={'Lorem ipsum dolor sit amet, consectetur adipisicing elit. Voluptas atque vel fuga illo cumque voluptatibus reprehenderit reiciendis commodi praesentium aperiam! In officiis architecto facere quas inventore illum laborum unde ratione!'} />)
  expect(component.find('LinkText').props().isShown).toBe(true);
});

