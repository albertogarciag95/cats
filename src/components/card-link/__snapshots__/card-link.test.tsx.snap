// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardLink match snapshot 1`] = `
<a
  className="sc-bdnxRM cDyIFN"
  onMouseEnter={[Function]}
>
  Read more
</a>
`;
